tests/unit_tests/test_channels_first_last.py

```diff
@@ -0,0 +1,75 @@
+import unittest
+
+import tensorflow as tf
+from src.models import model_name_to_class
+
+class TestCompareChannelsFirstLast(unittest.TestCase):
+  @classmethod
+  def setUpClass(cls):
+    cls.kwargs = {
+      'input_size': [224, 224, 3],
+      'changing_ids': [],
+      'num_classes': 10,
+    }
+
+    cls.list_of_models = model_name_to_class.values()
+
+    # below model do not work for upstride types
+    remove_models = [
+      # The channel_last and first values mismatch only for upstride types. 
+      # FIXME when this is prioritized.
+      "EfficientNetB0", 
+      "EfficientNetB1",
+      "EfficientNetB2",
+      "EfficientNetB3",
+      "EfficientNetB4",
+      "EfficientNetB5",
+      "EfficientNetB6",
+      "EfficientNetB7",
+      # SeparableConv2D is not supported for upstride types
+      "NASNetCIFAR",
+      "NASNetLarge",
+      "NASNetMobile",
+    ]
+
+    tmp_list_models = model_name_to_class
+    # remove models that are not supported
+    [tmp_list_models.pop(i) for i in remove_models]
```

Comment:
small thing, but notation here is not clear : i is generally for index
Here you should replace 'i' by 'model'

Reply:
updated accordingly.